packages/libs/eda/src/lib/core/components/filter/HistogramFilter.tsx

```diff
@@ -842,16 +835,33 @@ function tidyBinLabel(
 function formatStatValue(
   value: string | number | undefined,
   type: HistogramVariable['type']
-) {
+): string | number | string[] {
   if (value == null) return 'N/A';
-  return type === 'date'
-    ? String(value).replace(/T.*$/, '')
-    : // check a possible year variable
-    type === 'integer' && Number(value) >= 1900 && Number(value) <= 2100
-    ? Number.isInteger(value)
+
+  let formattedValue: string | number | string[] =
+    type === 'date'
+      ? String(value).replace(/T.*$/, '')
+      : // check a possible year variable
+      type === 'integer' && Number(value) >= 1900 && Number(value) <= 2100
+      ? Number.isInteger(value)
+        ? Number(value)
+        : Number(value).toFixed(4)
+      : // set conditions similar to plotly
```

Comment:
What do you think about breaking logic up into two steps? There is a lot of logic to keep track of here :D 

Possibly we could move the year check to a new step? It looks like some of that year logic is repeated in the new code as well?